src/test/java/com/amazon/ion/impl/EncodingDirectiveCompilationTest.java

```diff
@@ -619,8 +658,9 @@ private Macro writeSimonSaysMacro(IonRawWriter_1_1 writer) {
         );
     }
 
-    @Test
-    public void structAsParameter() throws Exception {
+    @ParameterizedTest
+    @EnumSource(InputType.class)
+    public void structAsParameter(InputType inputType) throws Exception {
```

Comment:
This is an example of a test that failed for `InputType.INPUT_STREAM` before the fix in IonCursorBinary.